src/home/NotificationCenter.tsx

```diff
@@ -161,6 +185,11 @@ export default function Notifications({ navigation }: NotificationsProps) {
   const handleViewableItemsChanged = useCallback(
     ({ viewableItems }: { viewableItems: ViewToken[] }) => {
       viewableItems.forEach(({ item }, index) => {
+        setNotificationPositions((prevNotificationPositions) => ({
+          ...prevNotificationPositions,
+          [item.id]: index,
+        }))
+
```

Comment:
I think we could avoid the extra state and the call to `setNotificationPositions` by directly consuming the `index` argument from the FlatList [renderItem](https://reactnative.dev/docs/flatlist#required-renderitem) method.

Added benefit is it avoids the re-render when items become visible.

But we'll have to wrap the `element` in a function call to pass the index.

What do you think?

Reply:
I'm doubtful. The `renderItem` receives the index within the *whole* data array, while we want to get the index only within currently *visible* items. It's because we want to know where the item was on screen when an interaction happened. But maybe I'm missing something?

Reply:
oh I see, yeah we need to agree on what this position is 👍 

To me it was the position from the whole list, not just on screen.
We have several similar instances of tracking positions, and so far they were the positions from the whole data.

For instance: 
https://github.com/valora-inc/wallet/blob/fc46ae9cc2641545d15d9321b59f80076f9edaac/src/analytics/Properties.tsx#L1197
https://github.com/valora-inc/wallet/blob/fc46ae9cc2641545d15d9321b59f80076f9edaac/src/analytics/Properties.tsx#L1189

Seeing https://github.com/valora-inc/wallet/blob/fc46ae9cc2641545d15d9321b59f80076f9edaac/src/home/NotificationCenter.tsx#L162-L173
To me this should have been the position from the whole list too.

I guess it depends what question we want to answer.
We should check with @ellenwoconnor.

In all cases we should make sure this isn't ambiguous. Maybe finding names that convey this more clearly?

Reply:
ah sorry i'm late to the party. i think the position we want to track is relative to the whole data - we want to learn if users are scrolling to the bottom of the list, rather than which position on the screen is most effective :)